### PR TITLE
build: remove fs-extra devdep

### PR DESCRIPTION
**Pull request by ckerr**
#### Description of Change

Node 20's `fs` module does everything we need now -- particularly with `{ recursive: true }` -- so we can drop our `fs-extra` dev dependency.

I noticed this while tracking down an intermittent test failure in our downstream fork.

CC @codebytere 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none.

### CODE REVIEW

File: script/gn-asar.js [hunk 1]
```diff
@@ -41,23 +41,15 @@ try {
...
+  asar.createPackageWithOptions(tmpPath, out[0], {});
 } catch (err) {
   console.error('Unexpected error while generating ASAR', err);
-  fs.remove(tmpPath)
-    .then(() => process.exit(1))
-    .catch(() => process.exit(1));
-  return;
+  process.exitCode = 1;
+} finally {
+  fs.rmSync(tmpPath, { force: true, recursive: true });
```
Comment (jkleinsc) [inline script/gn-asar.js:54]: fs.rmSync errors are not being caught here
```suggestion
 fs.promises.(tmpPath, { force: true, recursive: true })
     .then(() => process.exit(1))
    .catch(() => process.exit(1));
 return
```

File: script/spec-runner.js [hunk 2]
```diff
@@ -216,7 +216,7 @@ async function installSpecModules (dir) {
     env.npm_config_nodedir = path.resolve(BASE, `out/${utils.getOutDir({ shouldLog: true })}/gen/node_headers`);
   }
   if (fs.existsSync(path.resolve(dir, 'node_modules'))) {
-    await fs.remove(path.resolve(dir, 'node_modules'));
+    fs.rmSync(path.resolve(dir, 'node_modules'), { force: true, recursive: true });
```
Comment (jkleinsc) [inline script/spec-runner.js:219]: Why the change away from async here?

File: spec/api-autoupdater-darwin-spec.ts [hunk 3]
```diff
@@ -68,14 +68,14 @@ ifdescribe(shouldRunCodesignTests)('autoUpdater behavior', function () {
...
           appPJPath,
-          (await fs.readFile(appPJPath, 'utf8')).replace('1.0.0', version)
+          fs.readFileSync(appPJPath, 'utf8').replace('1.0.0', version)
         );
         const infoPath = path.resolve(secondAppPath, 'Contents', 'Info.plist');
-        await fs.writeFile(
+        fs.writeFileSync(
           infoPath,
-          (await fs.readFile(infoPath, 'utf8')).replace(/(<key>CFBundleShortVersionString<\/key>\s+<string>)[^<]+/g, `$1${version}`)
+          fs.readFileSync(infoPath, 'utf8').replace(/(<key>CFBundleShortVersionString<\/key>\s+<string>)[^<]+/g, `$1${version}`)
```
Comment (jkleinsc) [inline spec/api-autoupdater-darwin-spec.ts:78]: Why the change from async here?

**Review by jkleinsc — CHANGES_REQUESTED** [review]
The changes here fail to build.  I think instead of refactoring bits here we should stick to the orginal code as much as possible (eg don't change async to sync).

File: script/spec-runner.js (hunk 2 above)
Comment (ckerr) [reply script/spec-runner.js:219]: Here's a learning opportunity for me -- is there a difference between calling `await fs.promises.rm()` and `fs.rmSync()`? e.g. is one API more performant or more stable?

I assumed that sync calls were OK because the previous line uses `fs.existsSync()`, but maybe that's a special case because `fs.exists()` is deprecated?

File: spec/api-autoupdater-darwin-spec.ts (hunk 3 above)
Comment (ckerr) [reply spec/api-autoupdater-darwin-spec.ts:78]: Because the extra cognitive complexity of  `await fs.asyncFuncOne(await fs.asyncFuncTwo())` doesn't seem to buy us anything

File: script/gn-asar.js (hunk 1 above)
Comment (ckerr) [reply script/gn-asar.js:54]: They're not caught on line 62 of the original either

**Comment by ckerr** [comment]
Oh crud. I meant to delete a draft reply and instead deleted one of @jkleinsc's code review comments instead. Sorry about that, John. 

John's comment was that the new version of `gn-asar.js` does not catch an error thrown in `fs.rm()` and he's correct. In my defense, the same is true in the last line of the old version of the script too :smile_cat:

**Comment by ckerr** [comment]
> ckerr deleted a comment from jkleinsc now

Apparently I am just drunk with power and should put the keyboard away for the night

File: script/spec-runner.js (hunk 2 above)
Comment (ckerr) [reply script/spec-runner.js:219]: I'm of two minds on this.

OT1H, I agree with your comment that a procedural PR like this should KISS, e.g. avoid the refactors in `gn-asar`. IMO this outweighs the rest of the discussion. I'll revise to reduce diffs from `main`.

OT2H, this seems like a learning opportunity for me, so I have to ask -- is there a strong reason to preserve this `await fs.promises.foo()` call instead of calling `fs.fooSync()` when the pre-existing previous line is also making an fs sync call?

**Review by jkleinsc — APPROVED** [review]

File: script/spec-runner.js (hunk 2 above)
Comment (jkleinsc) [reply script/spec-runner.js]: @ckerr Because the function this is encapsulated in is async (async function installSpecModules), I'd lean towards using async, but the fact that there is already sync calls mixed in it doesn't make too much difference other than keeping the the updated code working the same way that the previous code did.

**Review by codebytere — APPROVED** [review]
